apis/crd/v1alpha1/cloudentityselector_webhook_test.go

```diff
@@ -0,0 +1,521 @@
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("same vpcMatch matchID abc and vmMatch matchName def" +
+				" configured"))
+		})
+		It("Validate different vpcMatch matchID and same vmMatch matchName in two vmSelectors", func() {
+			err = fakeClient.Create(context.Background(), account)
+			Expect(err).Should(BeNil())
+
+			selector = &CloudEntitySelector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testAccountNamespacedName.Name,
+					Namespace: testAccountNamespacedName.Namespace,
+				},
+				Spec: CloudEntitySelectorSpec{
+					AccountName: testAccountNamespacedName.Name,
+					VMSelector: []VirtualMachineSelector{
+						{
+							VpcMatch: &EntityMatch{
+								MatchID: "abc",
+							},
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "def",
+								},
+							},
+						},
+						{
+							VpcMatch: &EntityMatch{
+								MatchID: "xyz",
+							},
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "def",
+								},
+							},
+						},
+					},
+				},
+			}
+			err := selector.validateMatchSections()
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("Validate same vmMatch matchName(vpcMatch matchID in one) in two vmSelectors", func() {
+			err = fakeClient.Create(context.Background(), account)
+			Expect(err).Should(BeNil())
+
+			selector = &CloudEntitySelector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testAccountNamespacedName.Name,
+					Namespace: testAccountNamespacedName.Namespace,
+				},
+				Spec: CloudEntitySelectorSpec{
+					AccountName: testAccountNamespacedName.Name,
+					VMSelector: []VirtualMachineSelector{
+						{
+							VpcMatch: &EntityMatch{
+								MatchID: "abc",
+							},
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "def",
+								},
+							},
+						},
+						{
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "def",
+								},
+							},
+						},
+					},
+				},
+			}
+			err := selector.validateMatchSections()
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("Validate same vmMatch matchName in two vmSelectors", func() {
+			err = fakeClient.Create(context.Background(), account)
+			Expect(err).Should(BeNil())
+
+			selector = &CloudEntitySelector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testAccountNamespacedName.Name,
+					Namespace: testAccountNamespacedName.Namespace,
+				},
+				Spec: CloudEntitySelectorSpec{
+					AccountName: testAccountNamespacedName.Name,
+					VMSelector: []VirtualMachineSelector{
+						{
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "abc",
+								},
+							},
+						},
+						{
+							VMMatch: []EntityMatch{
+								{
+									MatchName: "abc",
+								},
+							},
+						},
+					},
+				},
+			}
+			err := selector.validateMatchSections()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("same vmMatch matchName abc configured in two match selectors"))
```

Comment:
make error msg constants in the pkg and use the same const in the test. this way, any change in error mesage will reflect at both places together

Reply:
Agree with you, but error messages do contain the configured(ex vmMatch matchName in this case) value too, so they cannot be constants.

Reply:
Done.